rs/tests/driver/src/driver/ic.rs

```diff
@@ -222,6 +245,15 @@ impl InternetComputer {
                 .collect();
             node.vm_resources = node.vm_resources.or(&self.default_vm_resources);
         }
+        for node in self.api_boundary_nodes.iter_mut() {
```

Comment:
Instead of this triplication, what about defining a method like:
```rust
    pub fn iter_mut_nodes(&self) -> ? /* I'm not sure about the correct return type */{
        self.subnets.iter_mut()
            .flat_map(|subnet|subnet.nodes.iter_mut())
            .chain(self.unassigned_nodes.iter_mut())
            .chain(self.api_boundary_nodes.iter_mut())
    }
```
and replacing these 3 for-loops with a single:
```rust
        for node in self.iter_mut_nodes() {
            node.required_host_features = node
                .required_host_features
                .iter()
                .chain(self.required_host_features.iter())
                .cloned()
                .collect();
            node.vm_resources = node.vm_resources.or(&self.default_vm_resources);
        }
```
You can also use this `iter_mut_nodes()` in the code for setting the node's `secret_key_store` and `ipv6` below.

Reply:
Or if that function is hard you could just inline into the for-loop like:

```rust
        for node in self.subnets.iter_mut().flat_map(|subnet|subnet.nodes.iter_mut().chain(self.unassigned_nodes.iter_mut()).chain(self.api_boundary_nodes.iter_mut()) {
            node.required_host_features = node
                .required_host_features
                .iter()
                .chain(self.required_host_features.iter())
                .cloned()
                .collect();
            node.vm_resources = node.vm_resources.or(&self.default_vm_resources);
        }
```

Reply:
That was just blind copy and paste without much thinking. Thanks a lot for the suggestion. It is indeed much, much nicer!